### PR TITLE
[8] Implement posted jobs page

### PR DESCRIPTION
**Pull request by priasdevo**
## What did you do

- image
- Placeholder of notibox, filter
- JobCard using infinite scroll view
- Connect JobCard with get/job API
- Onclick JobCard to JobDetail page
- Add color according to gender to gender-icon

## Demo

- https://dev.modela.miello.dev/job

### CODE REVIEW

File: apps/web/.env.prod [hunk 1]
```diff
@@ -1,3 +1 @@
-NEXT_PUBLIC_API_BASE_URL=https://api.modela.miello.dev
-NEXT_PUBLIC_ENABLE_PROXY=
-NEXT_PUBLIC_SITE_URL=
\ No newline at end of file
+NEXT_PUBLIC_API_BASE_URL=https://api-beta.modela.miello.dev
```
Comment (PKhing) [inline apps/web/.env.prod]: ?

File: apps/web/src/modules/job/list/components/FilterContainer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import {
+  Checkbox,
+  Divider,
+  FormControlLabel,
+  FormGroup,
+  Typography,
+} from '@mui/material'
+import React from 'react'
+
+import TextField from '../../../../../common/components/TextField'
```
Comment (PKhing) [inline apps/web/src/modules/job/list/components/FilterContainer/index.tsx:15]:
```suggestion
import TextField from 'common/components/TextField'
```

File: apps/web/src/modules/job/list/components/JobCardContainer/components/JobCard/index.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import { Divider, Typography } from '@mui/material'
+import { useRouter } from 'next/router'
+import React from 'react'
+
+import Footer from '../../../../../detail/components/JobCard/components/JobCardFooter'
+import Header from '../../../../../detail/components/JobCard/components/JobCardHeader'
```
Comment (PKhing) [inline apps/web/src/modules/job/list/components/JobCardContainer/components/JobCard/index.tsx:6]: Consider moving JobCardFooter and Header to a more common folder ex. `job/componnets` folder

File: apps/web/src/modules/job/list/components/JobCardContainer/components/JobCard/index.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    gender,
+    title,
+    wage,
+    jobId,
+    status,
+  } = prop
+  const router = useRouter()
+  const viewDetail = () => {
+    router.push(`/job/${jobId}`)
+  }
```
Comment (PKhing) [inline apps/web/src/modules/job/list/components/JobCardContainer/components/JobCard/index.tsx:26]: useCallback + consider moving this to hook file

File: apps/web/src/modules/job/list/components/JobCardContainer/index.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import { GetJobCardWithMaxPageDto } from '@modela/dtos'
+import React from 'react'
+
+import JobCard from './components/JobCard'
+import { CardBoxContainer } from './styled'
+
+export default function JobCardContainer(prop: GetJobCardWithMaxPageDto) {
+  const { jobs } = prop
+  console.log('SS')
+  console.log(jobs)
```
Comment (PKhing) [inline apps/web/src/modules/job/list/components/JobCardContainer/index.tsx:10]: remove this

File: apps/web/src/modules/job/list/components/JobCardContainer/index.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  return (
+    <CardBoxContainer>
+      {jobs.map((item, index) => {
+        return (
+          <div
+            key={`Job-${index}`}
```
Comment (PKhing) [inline apps/web/src/modules/job/list/components/JobCardContainer/index.tsx:16]: use `jobId` here

File: apps/web/src/modules/job/list/components/JobCardContainer/index.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+              companyName={item.companyName}
+              description={item.description}
+              castingImage={item.jobCastingImageUrl}
+              gender={item.gender}
+              actorCount={item.actorCount}
+              wage={item.wage}
+              dueDate={item.applicationDeadline}
+              jobId={item.jobId}
+              status={item.status}
+            />
```
Comment (PKhing) [inline apps/web/src/modules/job/list/components/JobCardContainer/index.tsx:36]:
```suggestion
            <JobCard {...item} />
```

File: apps/web/src/modules/job/list/components/NotiCardContainer/components/NotiCard/components/OfferDetail/index.tsx [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+export default function OfferDetail(prop: OfferDetailProps) {
+  const { offer } = prop
+
+  return (
+    <>
+      <Typography variant="subtitle2" sx={{ color: 'rgba(0,0,0,0.6)' }}>
+        {offer}
+      </Typography>
+    </>
```
Comment (PKhing) [inline apps/web/src/modules/job/list/components/NotiCardContainer/components/NotiCard/components/OfferDetail/index.tsx:14]:
```suggestion
      <Typography variant="subtitle2" sx={{ color: 'rgba(0,0,0,0.6)' }}>
        {offer}
      </Typography>
```

File: apps/web/src/modules/job/list/components/SearchBox/index.tsx [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import SearchOutlinedIcon from '@mui/icons-material/SearchOutlined'
+import React from 'react'
+
+import TextField from '../../../../../common/components/TextField'
```
Comment (PKhing) [inline apps/web/src/modules/job/list/components/SearchBox/index.tsx:4]:
```suggestion
import TextField from 'common/components/TextField'
```

File: apps/web/src/modules/job/list/pages/hooks/useJobListData/index.ts [hunk 10]
```diff
@@ -0,0 +1,38 @@
+import { GetJobCardWithMaxPageDto } from '@modela/dtos'
+import { apiClient } from 'common/utils/api'
+import { useRouter } from 'next/router'
+import { useState } from 'react'
+
+const useJobListData = () => {
+  const router = useRouter()
+  const [job, setJob] = useState<GetJobCardWithMaxPageDto>()
+  const [hasMore, setHasMore] = useState(true)
+  var pageControl: number = 1
```
Comment (PKhing) [inline apps/web/src/modules/job/list/pages/hooks/useJobListData/index.ts:10]: should we user `useRef` here ?

File: apps/web/src/modules/job/list/pages/hooks/useJobListData/index.ts [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+    try {
+      if (pageControl <= page) {
+        pageControl = page + 1
+        const res = (
+          await apiClient.get<GetJobCardWithMaxPageDto>(
+            `/job?limit=5&page=${page}`,
+          )
+        ).data
+        console.log(res)
+        console.log(res.maxPage, page, res.maxPage > page)
```
Comment (PKhing) [inline apps/web/src/modules/job/list/pages/hooks/useJobListData/index.ts:22]: remove this

File: apps/web/src/modules/job/list/pages/hooks/useJobListData/index.ts [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+  var pageControl: number = 1
+
+  const fetchData = async (page: number) => {
+    try {
+      if (pageControl <= page) {
+        pageControl = page + 1
+        const res = (
+          await apiClient.get<GetJobCardWithMaxPageDto>(
+            `/job?limit=5&page=${page}`,
+          )
```
Comment (PKhing) [inline apps/web/src/modules/job/list/pages/hooks/useJobListData/index.ts:19]:
```suggestion
          await apiClient.get<GetJobCardWithMaxPageDto>(
            `/job`, {
              params: {
                limit: 5,
                page
              }
            }
          )
```

File: apps/web/src/modules/job/list/pages/index.tsx [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+import NotiCardContainer from '../components/NotiCardContainer'
+import SearchBox from '../components/SearchBox'
+import useJobListData from './hooks/useJobListData'
+import { FilterBoxContainer, JobContainer, NotiContainer } from './styled'
+
+export default function JobList() {
+  var { job, hasMore, fetchData } = useJobListData()
+
+  return (
+    <>
```
Comment (PKhing) [inline apps/web/src/modules/job/list/pages/index.tsx:16]: remove `<></>`

File: apps/web/src/modules/job/list/pages/index.tsx [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+              pageStart={0}
+              loadMore={fetchData}
+              hasMore={hasMore}
+              loader={
+                <div className="loader" key={0}>
+                  Loading ...
+                </div>
+              }
+            >
+              {job ? <JobCardContainer {...job} /> : null}
```
Comment (PKhing) [inline apps/web/src/modules/job/list/pages/index.tsx:80]:
```suggestion
              {job && <JobCardContainer {...job} />}
```

File: apps/web/src/modules/job/list/pages/index.tsx [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+            }}
+          >
+            <InfiniteScroll
+              pageStart={0}
+              loadMore={fetchData}
+              hasMore={hasMore}
+              loader={
+                <div className="loader" key={0}>
+                  Loading ...
+                </div>
```
Comment (PKhing) [inline apps/web/src/modules/job/list/pages/index.tsx:77]: Consider using `<CircularProgress/>`

File: apps/web/src/modules/job/list/components/FilterContainer/index.tsx [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+      <Typography variant="body1"> ค่าจ้าง </Typography>
+      <TextField
+        fullWidth
+        placeholder="10000"
+        InputProps={{ endAdornment: <ArrowDropDownOutlined color="primary" /> }}
+      />
+
+      <Typography variant="subtitle2" sx={{ color: 'rgba(0,0,0,0.6)' }}>
+        {' '}
+        +- ไม่เกิน{' '}
```
Comment (PKhing) [inline apps/web/src/modules/job/list/components/FilterContainer/index.tsx:54]: Is `{' '}` necessary here ?

**Review by PKhing — CHANGES_REQUESTED** [review]
All `type.ts` file should be `types.ts` (ไม่เห็นตอนรีวิวของตุลย์ ชิบละ :joy:)

**Comment by PKhing** [comment]
👀 
image

**Comment by PKhing** [comment]
image
there should be more space before divider

File: apps/web/src/modules/job/list/components/FilterContainer/index.tsx [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+      <TextField
+        fullWidth
+        label="เวลาเริ่มต้นการถ่ายทำ"
+        InputProps={{ endAdornment: <Schedule color="primary" /> }}
+      />
+      <TextField
+        fullWidth
+        label="เวลาสิ้นสุดการถ่ายทำ"
+        InputProps={{ endAdornment: <Schedule color="primary" /> }}
+      />
```
Comment (miello) [inline apps/web/src/modules/job/list/components/FilterContainer/index.tsx:36]: This quite weird that we need to manual fill (maybe see this https://mui.com/x/react-date-pickers/time-picker/)

File: apps/web/src/modules/job/list/components/FilterContainer/index.tsx [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+
+export default function SearchBox() {
+  return (
+    <FilterBox>
+      <Typography variant="body1"> การถ่ายทำ </Typography>
+      <TextField
+        fullWidth
+        label="วันถ่ายทำ"
+        InputProps={{ endAdornment: <DateRangeOutlined color="primary" /> }}
+      />
```
Comment (miello) [inline apps/web/src/modules/job/list/components/FilterContainer/index.tsx:26]: Consider using date picker https://mui.com/x/react-date-pickers/date-picker/

File: apps/web/src/modules/job/list/components/FilterContainer/index.tsx [hunk 19]
```diff
@@ -0,0 +1,78 @@
...
+        {' '}
+        +- ไม่เกิน{' '}
+      </Typography>
+      <TextField
+        fullWidth
+        placeholder="1000"
+        InputProps={{ endAdornment: <ArrowDropDownOutlined color="primary" /> }}
+      />
+
+      <Divider variant="middle" style={{ width: '90%' }} />
```
Comment (miello) [inline apps/web/src/modules/job/list/components/FilterContainer/index.tsx:62]:
```suggestion
      <Divider variant="middle" sx={{ width: '90%' }} />
```

File: apps/web/src/modules/job/list/components/FilterContainer/index.tsx [hunk 20]
```diff
@@ -0,0 +1,78 @@
...
+      <Typography variant="body1"> สถานะการเปิดรับสมัคร </Typography>
+      <FormGroup>
+        <FormControlLabel control={<Checkbox />} label="เปิดรับสมัคร" />
+        <FormControlLabel control={<Checkbox />} label="ปิดรับสมัคร" />
+      </FormGroup>
+
+      <Divider variant="middle" style={{ width: '90%' }} />
```
Comment (miello) [inline apps/web/src/modules/job/list/components/FilterContainer/index.tsx:69]:
```suggestion
      <Divider variant="middle" sx={{ width: '90%' }} />
```

**Comment by PKhing** [comment]
image
คิดว่าบีบไป น่าจะต้องลงเป็นแบบ mobile ละ (~910 px)

File: apps/web/src/modules/job/list/components/NotiCardContainer/components/NotiCard/index.tsx [hunk 21]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+
+import Footer from './components/Footer'
+import Header from './components/Header'
```
Comment (miello) [inline apps/web/src/modules/job/list/components/NotiCardContainer/components/NotiCard/index.tsx:4]: Consider change file name to be more specific such as `NotiCardHeader`

File: apps/web/src/modules/job/list/pages/index.tsx [hunk 22]
```diff
@@ -0,0 +1,100 @@
...
+                  offer: '150k perk year',
+                  castingImage: '',
+                },
+                {
+                  title: 'Data Scientist offer',
+                  companyName: 'Google',
+                  offer: '200k perk year',
+                  castingImage: '',
+                },
+              ]}
```
Comment (miello) [inline apps/web/src/modules/job/list/pages/index.tsx:56]: Refactor to constants file

File: apps/web/src/modules/job/detail/components/JobCard/components/JobCardFooter/index.tsx [hunk 23]
```diff
@@ -1,5 +1,5 @@
-import { JobStatus, UserType } from '@modela/database'
-import { EventBusy, Money, PersonOutlined } from '@mui/icons-material'
+import { Gender, JobStatus, UserType } from '@modela/database'
```
Comment (miello) [inline apps/web/src/modules/job/detail/components/JobCard/components/JobCardFooter/index.tsx:1]:
```suggestion
import { Gender, JobStatus, UserType } from '@modela/dtos'
```

File: apps/web/src/modules/job/list/components/NotiCardContainer/components/NotiCard/components/Footer/index.tsx [hunk 24]
```diff
@@ -0,0 +1,23 @@
...
+  const read = () => {
+    window.alert('Readed')
+  }
+
+  return (
+    <FooterRow>
+      <Typography
+        variant="button"
+        color="primary"
+        style={{ cursor: 'pointer', justifyContent: 'left' }}
```
Comment (miello) [inline apps/web/src/modules/job/list/components/NotiCardContainer/components/NotiCard/components/Footer/index.tsx:16]:
```suggestion
        sx={{ cursor: 'pointer', justifyContent: 'left' }}
```

**Review by miello — CHANGES_REQUESTED** [review]

File: apps/web/src/modules/job/list/pages/hooks/useJobListData/index.ts (hunk 10 above)
Comment (priasdevo) [reply apps/web/src/modules/job/list/pages/hooks/useJobListData/index.ts:10]: do you refer to only this `var pageControl: number = 1` or all 4 line

Comment (PKhing) [reply apps/web/src/modules/job/list/pages/hooks/useJobListData/index.ts:10]: only `var pageControl`

File: apps/web/src/modules/job/list/components/NotiCardContainer/index.tsx [hunk 25]
```diff
@@ -0,0 +1,35 @@
...
+
+export default function NotiCardContainer(prop: CardArray) {
+  const { cardData } = prop
+
+  return (
+    <CardBoxContainer>
+      {cardData.map((item, index) => {
+        return (
+          <div
+            key={`Noti-${index}`}
```
[the remark this answers is not in the document]
Comment (priasdevo) [reply apps/web/src/modules/job/list/components/NotiCardContainer/index.tsx:15]: let wait for backend Implement first ~~

**Comment by priasdevo** [comment]
> image คิดว่าบีบไป น่าจะต้องลงเป็นแบบ mobile ละ (~910 px)

มันต้องเปลี่ยนประมาณเท่าไหร่นะ แบบตอนนี้ set ไว้ 900

**Comment by miello** [comment]
image

I think Divider here is quite too near to TextField

File: apps/web/src/modules/job/list/components/FilterContainer/style.ts [hunk 26]
```diff
@@ -0,0 +1,11 @@
...
+
+export const FilterBox = styled('div')`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  padding: 0px;
+  width: 100%;
+  min-width: 200px;
+  gap: 8px;
+`
```
Comment (miello) [inline apps/web/src/modules/job/list/components/FilterContainer/style.ts:11]: change filename from `style.ts` to styled.ts`

**Review by miello — CHANGES_REQUESTED** [review]
Last comment from me

**Review by miello — APPROVED** [review]

File: apps/web/src/modules/job/list/components/FilterContainer/index.tsx [hunk 27]
```diff
@@ -0,0 +1,128 @@
...
+import { FilterBox } from './style'
+
+export default function SearchBox() {
+  const [startShooting, setStartShooting] = React.useState<Dayjs | null>(null)
+  const [endShooting, setEndShooting] = React.useState<Dayjs | null>(null)
+  const [date, setDate] = React.useState<Dayjs | null>(null)
+  return (
+    <FilterBox>
+      <Typography variant="body1"> การถ่ายทำ </Typography>
+      <LocalizationProvider dateAdapter={AdapterDayjs}>
```
Comment (pisuttun) [inline apps/web/src/modules/job/list/components/FilterContainer/index.tsx:30]: Is this necessary? (I already include this it `_app.tsx` file )

Comment (priasdevo) [reply apps/web/src/modules/job/list/components/FilterContainer/index.tsx]: K forgot to read _app.tsx lol

**Review by PKhing — APPROVED** [review]

**Pull request merged**